src/traits.rs

```diff
@@ -513,3 +519,51 @@ pub trait WideningMul<Rhs = Self>: Sized {
     /// Perform widening multiplication.
     fn widening_mul(&self, rhs: Rhs) -> Self::Output;
 }
+
+/// A representation of an integer optimized for the performance of modular operations.
+pub trait Monty:
+    'static
+    + Clone
+    + Debug
+    + Eq
+    + Sized
+    + Send
+    + Sync
+    + Add<Output = Self>
+    + for<'a> Add<&'a Self, Output = Self>
+    + AddAssign
+    + for<'a> AddAssign<&'a Self>
+    + Sub<Output = Self>
+    + for<'a> Sub<&'a Self, Output = Self>
+    + SubAssign
+    + for<'a> SubAssign<&'a Self>
+    + Mul<Output = Self>
+    + for<'a> Mul<&'a Self, Output = Self>
+    + MulAssign
+    + for<'a> MulAssign<&'a Self>
+    + Neg<Output = Self>
+    + PowBoundedExp<Self::Integer>
+    + Square
+    + SquareAssign
+{
+    /// The original integer type.
+    type Integer: Integer<Monty = Self>;
+
+    /// The precomputed data needed for this representation.
+    type Params: Clone;
+
+    /// Create the precomputed data.
+    ///
+    /// Can return `None` if `modulus` is not valid for the representation;
+    /// see the documentation of the specific type for the requirements.
+    fn new_params(modulus: Self::Integer) -> CtOption<Self::Params>;
```

Comment:
Maybe `params_for_modulus`?

Reply:
I don't know, feels a little redundant? Fixed the docstring though since it's explicitly Montgomery representation.